modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java

```diff
@@ -140,9 +142,9 @@ public Application createArtifact(ApplicationDescriptor descriptor) throws IOExc
         applicationClassLoaderBuilderFactory.createArtifactClassLoaderBuilder();
     MuleDeployableArtifactClassLoader applicationClassLoader =
         artifactClassLoaderBuilder
-            .setDomain(domain)
+            .setDomainParentClassLoader((ArtifactClassLoader) domain.getArtifactClassLoader().getClassLoader().getParent())
```

Comment:
Is this logic something that should be resolved here or the builder should resolve it when build is invoked? See the comment regarding the jdoc method for the setDomainParentClassLoader which is not clear.

Reply:
I know there is a problem described in the ArtifactClassloader#getClassloader jdoc method, but isn't there a way to obtain this through a new method (even if we have to add this to the api, so this cast is needed here or in the builder). Not sure about that.

Reply:
the idea is for the api to be used in an external environemnt where you may have just the classloader for the domain, but not any of the domain structure. As this is, getting the classloader for a domain is an internal runtime thing to do, not something to bother external clients with.